File: 1.1/src/cz/mg/vulkan/PFNvkGetFenceStatus.java
package cz.mg.vulkan;

public class PFNvkGetFenceStatus extends VkFunctionPointer {
    public PFNvkGetFenceStatus() {
    }

    protected PFNvkGetFenceStatus(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetFenceStatus(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetFenceStatus(long value) {
        setValue(value);
    }

    public PFNvkGetFenceStatus(VkInstance instance) {
        super(instance, new VkString("vkGetFenceStatus"));
    }

    public void call(VkDevice device, VkFence fence, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), fence != null ? fence.getVkAddress() : VkPointer.getNullAddressNative(), rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long fence, long rval);


    public int call(VkDevice device, VkFence fence){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), fence != null ? fence.getVkAddress() : VkPointer.getNullAddressNative());
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, long fence);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetFenceStatus.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetFenceStatus_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong fence, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetFenceStatus f = (PFN_vkGetFenceStatus)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkFence*)jniLongToPointer(fence))
    );
}


jint Java_cz_mg_vulkan_PFNvkGetFenceStatus_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong fence){
    (void)env;
    (void)jc;
    PFN_vkGetFenceStatus f = (PFN_vkGetFenceStatus)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkFence*)jniLongToPointer(fence))
    );
}

